src/plugins/GLesmos/glesmosCanvas.ts

```diff
@@ -112,27 +160,128 @@ void main() {
 export function initGLesmosCanvas() {
   //================= INIT ELEMENTS =======================
   let c: HTMLCanvasElement = document.createElement("canvas");
-  let gl: WebGLRenderingContext = c.getContext("webgl2", {
+  let gl: WebGL2RenderingContext = c.getContext("webgl2", {
     // Disable premultiplied alpha
     // Thanks to <https://stackoverflow.com/a/12290551/7481517>
     premultipliedAlpha: false,
-  }) as WebGLRenderingContext;
+  }) as WebGL2RenderingContext;
+
+  let prevFramebuffer: WebGLFramebuffer | null;
+  let prevTexture: WebGLTexture | null;
+
+  let currFramebuffer: WebGLFramebuffer | null;
+  let currTexture: WebGLTexture | null;
+
+  let currFramebuffer2: WebGLFramebuffer | null;
+  let currTexture2: WebGLTexture | null;
```

Comment:
What are the relations between these nulls? If the buffer is null iff the texture is null, then they can be merged into a single type, so they require only a single null check and the relationship is clear.
```ts
type MaybeFrameData = FrameData | null
interface FrameData {
  buffer: WebGLFrameBuffer;
  texture: WebGLTexture;
};
```

Reply:
This was more or less a hack I used before I knew how to properly use TypeScript's type system. Now I've decided to throw an error if the functions that create these return `null` because that really shouldn't happen during execution.